pkg/rekor/rekor.go

```diff
@@ -43,21 +42,25 @@ import (
 	"github.com/sigstore/sigstore/pkg/cryptoutils"
 )
 
+// Verifier represents a mechanism to get and verify Rekor entries.
 type Verifier interface {
-	Get(ctx context.Context, commit string, cert *x509.Certificate) (*models.LogEntryAnon, error)
+	Get(ctx context.Context, data []byte, cert *x509.Certificate) (*models.LogEntryAnon, error)
```

Comment:
This `rekor.Verifier` interface doesn't seem very specific to gitsign. If it was verifying commit SHAs specifically that'd make sense. Otherwise, maybe this belongs in sigstore/sigstore (or sigstore/rekor) so others can use it.

OTOH, I'm a big fan of not exporting something, in here or in broader sigstore, until there's a client, so you know what you need from it. If you have a client and this is the interface that makes sense there, this sounds fine to me.

gitsign's Go package API compatibility promises seem quite a bit more lax than sigstore/sigstore's, so if we need to change this later (or unexport it) I think we can.

Reply:
wdyt about slimming this down to:

```go
type Verifier interface {
	Verify(ctx context.Context, commitSHA string, cert *x509.Certificate) (*models.LogEntryAnon, error)
}
```
?

The main reason why this was split was so that we could tell the difference between "there was no tlog entry" and "we couldn't validate the tlog entry we found". But we could also surface this type of information within the error itself.

Reply:
Went ahead and made this change. lmk what you think!